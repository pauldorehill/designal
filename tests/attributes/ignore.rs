use designal::Designal;

#[derive(Designal)]
#[designal(ignore)]
struct HumanBean();

#[derive(Designal)]
struct HumanBean1 {
    #[designal(ignore)]
    #[designal(ignore)]
    taste: String,
}
#[derive(Designal)]
struct HumanBean2 {
    #[designal(ignore)]
    #[designal(remove)]
    taste: String,
}

#[derive(Designal)]
struct HumanBean3 {
    #[designal(ignore)]
    #[designal(rename = "crunch")]
    taste: String,
}

#[derive(Designal)]
struct HumanBean4 {
    #[designal(ignore)]
    #[designal(keep_rc)]
    taste: String,
}

#[derive(Designal)]
struct HumanBean5 {
    #[designal(ignore)]
    #[designal(keep_arc)]
    taste: String,
}

fn main() {}
