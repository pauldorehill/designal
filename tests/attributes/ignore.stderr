error: Ignore is not valid at the container level
 --> $DIR/ignore.rs:4:12
  |
4 | #[designal(ignore)]
  |            ^^^^^^

error: You should only ignore once
  --> $DIR/ignore.rs:10:16
   |
10 |     #[designal(ignore)]
   |                ^^^^^^

error: You are ignoring designal on this field, but have added other attributes
  --> $DIR/ignore.rs:15:16
   |
15 |     #[designal(ignore)]
   |                ^^^^^^

error: You are ignoring designal on this field, but have added other attributes
  --> $DIR/ignore.rs:22:16
   |
22 |     #[designal(ignore)]
   |                ^^^^^^

error: You are ignoring designal on this field, but have added other attributes
  --> $DIR/ignore.rs:29:16
   |
29 |     #[designal(ignore)]
   |                ^^^^^^

error: You are ignoring designal on this field, but have added other attributes
  --> $DIR/ignore.rs:36:16
   |
36 |     #[designal(ignore)]
   |                ^^^^^^
